akka-remote/src/main/scala/akka/remote/AckedDelivery.scala

```diff
@@ -86,12 +86,14 @@ class ResendUnfulfillableException
  * @param nacked   Sequence of messages that has been explicitly negative acknowledged.
  * @param maxSeq The maximum sequence number that has been stored in this buffer. Messages having lower sequence number
  *               will be not stored but rejected with [[java.lang.IllegalArgumentException]]
+ * @param hasBeenUsed Indicates whether the buffer has been used at all
  */
 case class AckedSendBuffer[T <: HasSequenceNumber](
   capacity: Int,
   nonAcked: IndexedSeq[T] = Vector.empty[T],
   nacked: IndexedSeq[T] = Vector.empty[T],
-  maxSeq: SeqNo = SeqNo(-1)) {
+  maxSeq: SeqNo = SeqNo(-1),
+  hasBeenUsed: Boolean = false) {
```

Comment:
this is unfortunately not internal, so we have to be creative around binary compatibility issue